akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala

```diff
@@ -218,11 +219,26 @@ final class EventEnvelope[Event](
   /**
    * Java API
    */
+  @deprecated("Use getEventMetadata with metaDataType parameter")
   def getEventMetaData(): Optional[AnyRef] = {
     import scala.jdk.OptionConverters._
+    // FIXME for compatibility this would only look for ReplicatedEventMetadata and extract that
+    // from composite metadata
     eventMetadata.map(_.asInstanceOf[AnyRef]).toJava
   }
 
+  /**
+   * Java API: The metadata of a given type that is associated with the event.
+   */
+  def getEventMetadata[M](metaDataType: Class[M]): Optional[M] =
```

Comment:
Not important, but consistently use `metadata` as one word, with `metadataType`?